internal/pkg/manifest/validate.go

```diff
@@ -1721,6 +1738,52 @@ func validateDepsForEssentialContainers(deps map[string]containerDependency) err
 	return nil
 }
 
+func validateExposedPorts(opts validateExposedPortsOpts) error {
```

Comment:
can we potentially simplify by leveraging the new `ExposedPort` method on all types?

```go
if err := validateExposedPorts(mft.ExposedPorts(), &mft.RoutingRule); err != nil  { return err }

func validateExposedPorts(ports []ExposedPort, http *RoutingRuleConfiguration) error {
  // (1) check duplicate in ports
  
  // (2) check if http.targetPort and http.targetContainer are compatible
}
```

Reply:
I think the current implementation is easier for us validate, because we are validating against the manifest inputs. Once we call `ExposePorts()` then there is only `ExposedPort` objects and to validate those objects for duplicates and `http.targetPort` and `http.targetContainer` are compatible use case will be same amount of code and effort. Instead current validation keeps it easy. We just need to call  `validateExposedPorts` with right inputs from all the types of services.